Вторая часть курсового проекта.

aopanasenko opened the pull request:
Рассчитаны положение спутника в заданный момент времени, траектории его орбиты в системе EСI и ECEF. Cмущает вид траекторий, так как проходят через центр Земли.

aopanasenko committed:
Вторая часть курсового проекта.

Рассчитаны положение спутника в заданный момент времени, траектории его орбиты в системе EСI и ECEF. Cмущает вид траекторий, так как проходят через центр Земли.

Korogodin commented:
Вангую проблему тут:
Vk = atan(((sqrt(1-e^2)*sin(Ek))/(1 - e*cos(Ek))) / ((cos(Ek) - e)/(1 - e*cos(Ek))));

Воспользуйся функцией atan2

aopanasenko committed:
Курсовой проект. Часть 2 (Исправление)

Исправил функцию atan() на atan2(,) - помогло. SkyView получился совсем не таким, каким нужно. Возможна ли ошибка в просчете координат?

aopanasenko commented:
Залил исправленную версию. Изменил atan() на atan2(,) - c построениями траекторий спутников это помогло. В ИКД GPS указаны величины, которые должны иметь размерность полуциклы/c, а не рад/сек. Правильно ли я понял, что для перевода из градусов в полуциклы нужно умножить на 0.5/180, так как 180 градусов это и есть полуцикл?

Korogodin commented:
> В ИКД GPS указаны величины, которые должны иметь размерность полуциклы/c, а не рад/сек. Правильно ли я понял, что для перевода из градусов в полуциклы нужно умножить на 0.5/180, так как 180 градусов это и есть полуцикл?

Про какие величины речь? Мне казалось, все необходимые параметры получаются из таблички RTKLIB, там размерность подписана. 

По отчету: проблема с SkyView. В коде ошибка в глаза не бросается. Предлагаю построить Elevation от времени для проверки.

aopanasenko commented:
1. Я имел ввиду величины, через которые рассчитываются координаты спутника. В ИКД некоторые из них имеют размерность в полуциклах (на фото во вложении), а в RTKLIB - в градусах. Собственно я рассчитывал положения спутника, переводя все в радианы. Нужно ли переводить в полуциклы согласно ИКД? 
2. Так же прикладываю график зависимости Elevation от времени. Его вид не очень удовлетворяет ожидаемому: по идее график функции Elevation(t) должен монотонно возрастать до определенного значения, а после убывать.
![Параметры в ИКД](https://user-images.githubusercontent.com/35809906/57038790-76128c80-6c63-11e9-8c7d-2fdf660e23aa.png)
![image](https://user-images.githubusercontent.com/35809906/57038955-ea4d3000-6c63-11e9-88e1-f9de24af71d9.png)
![image](https://user-images.githubusercontent.com/35809906/57038987-06e96800-6c64-11e9-81d8-1b8043475280.png)

Korogodin commented:
1. Эфемериды, т.е. параметры модели движения спутника, проходят длинную цепочку, перед тем как попасть в твой скрипт. Система контроля и управления измеряет положение аппаратов, математическая модель с порядка 3000 параметров подгоняется под измерения, затем на неё натягивается модель, которую ты используешь. Эти параметры огрубляются, передаются на борт, он потом закладывает их в сигнал. Приемник принимает сигнал, демодулирует. Раскладывает по кадрам. Заворачивает в пакет 0xF7. Пакеты дампятся, затем конвертируются с помощью парсера nvs.c в RTKLIB, затем выводятся в таблице. 

Практически на каждом шаге меняется ни только размерность, но представление этих чисел. 

То, что написано в ИКД и приведено в таблице выше, представление в кадре навигационного сообщения Legacy NAV сигнала GPS L1C/A и имеет очень опосредованное отношение к таблице в RTKLIB.  

2. Попробуй представить положение спутника с углом места 150 градусов.

aopanasenko committed:
Курсовой проект. Часть 2. Исправление № 2.

Получены нормальные графики SkyView, а так же зависимость угла места от времени.

aopanasenko committed:
Add files via upload

Развернул оси на графике SkyView.

Korogodin commented:
У вас у всех повторяется одна проблема: вместо того, чтоб на polar развернуть ось и построить elevation, вы строите 90-elevation. В итоге кривая SkyView верная, а подпись на оси на графике - нет. 

Второй этап принимаю
